### PR TITLE
Add IO.randomUUID convenience method

### PR DESCRIPTION
**Pull request by Daenyth**

### CODE REVIEW

**Review by armanbilge — DISMISSED** [review]

File: core/shared/src/main/scala/cats/effect/IO.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ import cats.{
 }
 import cats.data.Ior
 import cats.effect.instances.spawn
-import cats.effect.std.{Console, Env}
+import cats.effect.std.{Console, Env, GenUUID}
```
Comment (djspiewak) [inline core/shared/src/main/scala/cats/effect/IO.scala]: I believe you mean `UUIDGen` :-)

**Review by djspiewak — CHANGES_REQUESTED** [review]
@Daenyth Looks like this needs to be split

File: core/shared/src/main/scala/cats/effect/IO.scala (hunk 1 above)
Comment (Daenyth) [reply core/shared/src/main/scala/cats/effect/IO.scala]: Damnit

**Comment by wjoel** [comment]
Wouldn't `Random` be a better place for this? https://github.com/typelevel/cats-effect/blob/series/3.x/std/shared/src/main/scala/cats/effect/std/Random.scala

**Comment by djspiewak** [comment]
It could be, yes, but this is also a wildly common case, so I think it's fair to lean a bit toward better convenience. `Random` isn't any harder to summon than `UUIDGen`, but IO is just… there.

**Comment by wjoel** [comment]
Fair enough (even if I personally don't generate random UUIDs *that* often ;) ), but maybe the help text could include a reference `Random` so newbies that type `IO.rando...` have a better chance of discovering it. Just a thought.

**Review by djspiewak — APPROVED** [review]

**Comment by Daenyth** [comment]
I was looking at that too - `UUIDGen` doesn't actually use `Random`, it just wraps `UUID.randomUUID()` in `Sync[F].delay`

**Comment by djspiewak** [comment]
This actually might be something that should be addressed, since randomUUID can block threads.

**Comment by rossabaker** [comment]
I have seen `UUID.randomUUID` kill more than one service in the years BCE (Before Cats Effect).  The synchronization isn't obvious until you start profiling why you're getting single-threaded performance.

**Pull request merged**